### DISCUSSION TITLE
How to setup clojure_lsp correctly?

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#4697] [Q&A] [answered]
I installed `clojure_lsp` via `mason`, and then configure it through `nvim-lspconfig`:

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        clojure_lsp = {},
      },
  },
```
And this LSP is detected in a clj file (`LspInfo`):

<img width="896" alt="image" src="https://github.com/user-attachments/assets/9abb4b55-6530-489d-b79b-07717005bf27">

However, the auto-completion provided by LSP does not work. Is it a bug caused by this LSP, or my config is wrong?

**Comment by Jacobus-afk** [comment]
would your use case work by enabling it in [LazyExtras](http://www.lazyvim.org/extras/lang/clojure)?

**Reply by ChenZhongPu** [reply]
Yes, I have also enabled it in `LazyExtras`. In fact, the `clojure` in `LazyExtras` is for REPL, not LSP. So I also installed [clojure_lsp](https://github.com/clojure-lsp/clojure-lsp).

BTW, I am new to Clojure, and this is my first day to program with it.

**Reply by Jacobus-afk** [reply]
oh, i see.. sorry also not at all familiar with clojure..

it looks like `cmp-conjure` is the autocompletion plugin. could you also try installing it with mason. 

then with the clojure lazyextra enabled, it will hopefully work

**Reply by ChenZhongPu** [reply]
@Jacobus-afk Thanks. I found the solution.

**Comment by ChenZhongPu** [comment, marked as answer]
I found that `clojure_lsp` works only when there is root file.

> root_dir = util.root_pattern('project.clj', 'deps.edn', 'build.boot', 'shadow-cljs.edn', '.git', 'bb.edn'),

So, I created an **empty** `deps.edn` to indicate this is a project. Now it works as expected!

<img width="798" alt="image" src="https://github.com/user-attachments/assets/4e388d8b-a65b-45bc-bff5-2a71624c2e70">
